hat/assets/js/apps/Iaso/domains/instances/components/CreateReAssignDialogComponent.tsx

```diff
@@ -0,0 +1,158 @@
+
+type Props = {
+    titleMessage: any;
+    confirmMessage: any;
+    cancelMessage: any;
+    formType: {
+        id?: number;
+        periodType: string;
+    };
+    currentInstance?: {
+        id: number;
+        period?: string;
+        // eslint-disable-next-line camelcase
+        org_unit?: any;
+    };
+    onCreateOrReAssign: (
+        // eslint-disable-next-line no-unused-vars
+        instanceOrForm: Instance | { id: number },
+        // eslint-disable-next-line no-unused-vars,camelcase
+        payload: { period: any; org_unit: any },
+    ) => void;
+    orgUnitTypes: number[];
+    isOpen: boolean;
+    closeDialog: () => void;
+};
+
+export const CreateReAssignDialogComponent: FunctionComponent<Props> = ({
+    titleMessage,
+    confirmMessage = MESSAGES.ok,
+    cancelMessage = MESSAGES.cancel,
+    formType,
+    currentInstance,
+    onCreateOrReAssign,
+    orgUnitTypes = [],
+    isOpen,
+    closeDialog,
+}) => {
+    const { formatMessage } = useSafeIntl();
+    const [fieldValue, setFieldValue] = useState(() => {
+        let initialPeriod: string | undefined;
+        let initialPeriodErrors: string[] = [];
+        if (currentInstance) {
+            initialPeriod = currentInstance.period;
+            const isOriginalPeriodValid = isValidPeriod(
+                initialPeriod,
+                formType.periodType,
+            );
+            if (!isOriginalPeriodValid) {
+                initialPeriodErrors.push(
+                    formatMessage(MESSAGES.initialPeriodError, {
+                        period: currentInstance.period,
+                    }),
+                );
+            }
+            return {
+                orgUnit: { value: currentInstance.org_unit, errors: [] },
+                period: { value: initialPeriod, errors: initialPeriodErrors },
+            };
+        } else if (formType.periodType) {
+            // On creation make a default period
+            const toDay = new Date();
+            const period = new Period(
+                toDay.getFullYear() + `0${toDay.getMonth() + 1}`.slice(-2),
+            );
+            initialPeriod = period.asPeriodType(
+                formType.periodType,
+            ).periodString;
+        }
+        return {
+            orgUnit: { value: currentInstance?.org_unit, errors: [] },
+            period: { value: initialPeriod, errors: initialPeriodErrors },
+        };
+    });
+    const isPeriodRequired = Boolean(formType.periodType);
+    const allowConform =
+        Boolean(fieldValue.orgUnit.value) &&
+        isPeriodRequired &&
+        Boolean(fieldValue.period.value);
+    // TODO Above logic should be moved to Formik
+
+    const onConfirm = () => {
+        const currentFormOrInstanceProp = currentInstance || formType;
+        // ignore ts error for currentFormOrInstanceProp as the logic ensure there is one
+        // open to cleaner option
+        // @ts-ignore
+        onCreateOrReAssign(currentFormOrInstanceProp, {
```

Comment:
your problem is on the id that can be undefined here:
![Screenshot 2022-12-01 at 16 33 40](https://user-images.githubusercontent.com/12494624/205094087-c8c2eea2-0605-48cb-95b4-342684c7af5f.png)